--- SFST/src/fst.C~	2012-02-20 13:34:34.000000000 +0100
+++ SFST/src/fst.C	2013-12-03 08:43:28.287475829 +0100
@@ -511,11 +511,11 @@
     for( ArcsIter it(node->arcs()); it; it++ ) {
       Arc *arc=it;
 
-      NodeHashSet::iterator it=previous.insert(node).first;
+      NodeHashSet::iterator itp=previous.insert(node).first;
       path.push_back(arc->label());
       enumerate_paths_node( arc->target_node(), path, previous, result );
       path.pop_back();
-      previous.erase(it);
+      previous.erase(itp);
     }
   }
 
